hastory-server/src/Data/Hastory/Server.hs

```diff
@@ -91,28 +101,30 @@ tokenLength = 20
 
 -- | Generate a random token that this server requires with every request.
 -- See @_ssToken@ for more info.
-generateToken :: MonadIO m => m T.Text
-generateToken = T.pack . take tokenLength . randomRs ('a', 'z') <$> liftIO newStdGen
+generateToken :: MonadIO m => m Token
+generateToken = Token . T.pack . take tokenLength . randomRs ('a', 'z') <$> liftIO newStdGen
 
 -- | Displays the port this server will use. This port is configurable via command-line flags.
 reportPort :: MonadLogger m => Options -> m ()
 reportPort Options {..} = logInfo $ "Starting server on port " <> T.pack (show _oPort)
 
--- | Logs information about the data file. This data file serves as a database for this primitive append-only server.
-reportDataFileStatus :: (MonadIO m, MonadLogger m) => Options -> m ()
-reportDataFileStatus Options {..} = do
-  dataFileExists <- liftIO $ fileExist _oDataOutputFilePath
-  if dataFileExists
-    then logInfo $
-         "Data file exists at " <> T.pack _oDataOutputFilePath <> ". Appending commands to it."
-    else logInfo "Data file doesn't exist. Creating a new one."
-
 -- | Starts a webserver by reading command line flags.
-hastoryServer :: (MonadIO m, MonadLogger m) => m ()
+hastoryServer :: (MonadIO m, MonadLogger m, MonadUnliftIO m, MonadThrow m) => m ()
 hastoryServer = do
   options@Options {..} <- liftIO $ A.execParser optParser
   reportPort options
-  reportDataFileStatus options
-  token <- generateToken
-  logInfo $ "Token: " <> token
-  liftIO $ Warp.runSettings (mkWarpSettings options) (app options (ServerSettings token))
+  token   <- generateToken
+  dir     <- getAppUserDataDir "hastory-server"
+  relFile <- parseRelFile "server.db"
+  dbPool  <- prepareDb (dir </> relFile)
+  let (Token token') = token
+  logInfo $ "Token: " <> token'
+  liftIO $ Warp.runSettings (mkWarpSettings options) (app options (ServerSettings token dbPool))
+
+prepareDb ::
+     (MonadIO m, MonadLogger m, MonadUnliftIO m) => Path Abs File -> m (Pool SqlBackend)
+prepareDb file = do
+  _    <- ensureDir (parent file)
+  pool <- SQL.createSqlitePool (T.pack $ toFilePath file) 10
+  _    <- SQL.runSqlPool (SQL.runMigration migrateAll) pool
```

Comment:
If you want to get rid of the logging during migrations, you have to use `runMigrationSilent` or something (I can't remember).